src/stopping/electron_range.cpp

```diff
@@ -52,24 +50,79 @@ nb::object electron_range(nb::object input, nb::object material, nb::object mode
             throw std::runtime_error(error_msg.c_str()); // Nanobind will translate this
         }
     }
+    return material_id;
+}
 
+int process_model(const nb::object& model){
     int model_id = 0;
     if (nb::isinstance<nb::str>(model)) {
-        std::string model_name = nb::cast<std::string>(model);
-         try {
+        auto model_name = nb::cast<std::string>(model);
+        try {
             model_id = get_model_id(model_name);
-         } catch (const std::runtime_error &e) {
-             throw nb::value_error(e.what());
-         }
+        } catch (const std::runtime_error &e) {
+            throw nb::value_error(e.what());
+        }
     } else if (nb::isinstance<nb::int_>(model)) {
         model_id = nb::cast<int>(model);
     } else {
         throw nb::type_error("Model argument must be either an integer or a string");
     }
+    return model_id;
+}
+bool check_int_dtype(const nb::object& array){
```

Comment:
that may be also used in some other places, consider moving it to `wrapper_template.h` or some other central place